Framework/Geometry/inc/MantidGeometry/Rendering/CacheGeometryGenerator.h

```diff
@@ -40,15 +40,19 @@ namespace Geometry {
    File change history is stored at: <https://github.com/mantidproject/mantid>
 */
 class CSGObject;
+class MeshObject;
+
 class MANTID_GEOMETRY_DLL CacheGeometryGenerator {
 private:
-  CSGObject *Obj;     ///< Input Object
-  int mNoOfVertices;  ///< number of vertices
-  int mNoOfTriangles; ///< number of triangles
-  double *mPoints;    ///<double array or points
-  int *mFaces;        ///< Integer array of faces
+  CSGObject *csgObj;   ///< Input CSGObject
```

Comment:
What is the reason that we need to keep 2 pointers here rather than a single `IObject` pointer?

Reply:
We need to keep 2 pointers, because behaviour is different for MeshObject than CSGObject, see the code for `CacheGeometry::Generate() `  (line 44 of cpp file) for example.